substrate-node/pallets/pallet-smart-contract/src/tests.rs

```diff
@@ -722,15 +747,15 @@ fn test_node_contract_billing_details() {
 
         push_nru_report_for_contract(1, 10);
 
-        let contract_to_bill = SmartContractModule::contract_to_bill_at_block(10);
+        let contract_to_bill = SmartContractModule::contract_to_bill_at_block(0);
         assert_eq!(contract_to_bill, [1]);
 
         let initial_total_issuance = Balances::total_issuance();
         // advance 25 cycles
         for i in 0..25 {
             pool_state
                 .write()
-                .should_call_bill_contract(1, 10 + i * 10, Ok(()));
+                .should_call_bill_contract(1, Ok(Pays::Yes.into()));
```

Comment:
I just have one concern now that block_number has been removed from the transactional call bill_contract. It's not a huge concern but we no longer know for sure at which block_number the contract was billed. For example here we don't know for sure that the contract was billed at block 10, 20, etc. We do know it was billed within the past 10 blocks. So lets say the contract was billed at block 8 while we expect it to be billed at 10. This function will now not notice the difference while in some cases it may be important. It could be a good idea to extend the pool with an extra variable block_number where that variable defines when the call should actually take place. It can be optional too.